src/main/java/me/xmrvizzy/skyblocker/skyblock/FairySouls.java

```diff
@@ -147,12 +154,12 @@ private static boolean isFairySoulNotFound(BlockPos fairySoul) {
         if (foundFairiesForProfileAndLocation == null) {
             return true;
         }
-        return !foundFairiesForProfileAndLocation.contains(fairySoul);
+        return !foundFairiesForProfileAndLocation.contains(fairySoulPos);
     }
 
     public static void onChatMessage(Text text, boolean overlay) {
         String message = text.getString();
-        if (message.equals("You have already found that Fairy Soul!") || message.equals("SOUL! You found a Fairy Soul!")) {
+        if (message.equals("You have already found that Fairy Soul!") || message.equals("§d§lSOUL! §fYou found a §dFairy Soul§f!")) {
```

Comment:
Are you sure about this change? `getString()` shouldn't have the formatting codes?